scripts/prepare-coverage.bat

```diff
@@ -0,0 +1,7 @@
+if not defined PYTEST_NO_COVERAGE (
+    set "_PYTEST_TOX_COVERAGE_RUN=coverage run --source {envsitepackagesdir}/_pytest/,%CD%/testing -m"
+    set "_PYTEST_TOX_EXTRA_DEP=coverage-enable-subprocess"
```

Comment:
i beleivethis should add coverage in any case

Reply:
no, the requirement, i beleive i saw a few errors about the coverage binary being not in the env

Reply:
also i believe coverage integration should be a tox pugin -  @tox-dev might have an oppinion there

Reply:
Are you sure it was `coverage`? I see `tox` warning about `env`:

```
WARNING: test command found but not installed in testenv
  cmd: /usr/bin/env
  env: /home/travis/build/pytest-dev/pytest/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
```

I've tried to fix this and the py27-xdist job by explicitly setting `COVERAGE_FILE` and `COVERAGE_PROCESS_START` outside tox in a60cc256d3512c9663b1356cb19bbf7909e507b9, but this ended up breaking other things so I removed this commit from this branch for the moment.

Let's see what @blueyed advises here. 👍

Reply:
@nicoddemus oh, then i did missread - i beleive its valid to whitelist ENV

Reply:
> I beleive its valid to whitelist ENV

I don't mind too, but I think a cross-platform solution like in a60cc256d3512c9663b1356cb19bbf7909e507b9 would be better for maintainability, we would be using the same commands and environment variables in both platforms.

Reply:
This will also collect coverage for tox then already - I have seen errors that source for `/tmp/pip-req…` was missing.  It would need explicit includes then probably.

Instead of using `coverage` we could have a wrapper script instead that sets it then?

Reply:
Re `tox-coverage`: I think that would be nice to have, and could look similar to https://github.com/tarpas/pytest-testmon/blob/master/testmon/tox_testmon.py - i.e. it would automatically install codecov and set up env, if `tox --coverage` was used.

Reply:
> Instead of using coverage we could have a wrapper script instead that sets it then?

See https://github.com/pytest-dev/pytest/pull/3923#discussion_r214698035, I don't think that's necessary, but would indeed be a good solution if that was not the case. 👍